benchmark/aws/cloudformation.py

```diff
@@ -0,0 +1,107 @@
+    
+    @property
+    def id(self):
+        if hasattr(self, '_stack_id'):
+            return self._stack_id
+    
+    @property
+    def created(self):
+        if hasattr(self, '_created'):
+            return self._created
+    
+    @property
+    def status(self):
+        self._describe_stack()
+        if hasattr(self, '_status'):
+            return self._status
+        
+    @property
+    def resources(self):
+        self._stack_resources()
+        return self._resources
+        
+    def _describe_stack(self):
+        try:
+            stack_details = self._client.describe_stacks(StackName=self._name)
+            self._status = stack_details['Stacks'][0]['StackStatus']
+        except botocore.exceptions.ClientError as exp:
+            self.logger.exception(f'Got following error while triggering describe_stacks {exp}')
+        except Exception as exp:
+            self.logger.exception(f'Got following error while triggering describe_stacks {exp}')
+    
+    def _stack_resources(self):
+        try:
+            stack_resources = self._client.describe_stack_resources(StackName=self._name)
+            self._resources = {resource['LogicalResourceId']: resource['PhysicalResourceId'] 
+                               for resource in stack_resources['StackResources']}
+        except Exception as exp:
+            self.logger.exception(f'Got following error while triggering describe_stack_resources {exp}')
+               
+    def wait(self, waiter_name):
+        self._client_wrapper.waiter = waiter_name
+        with TimeContext(f'Waiting for `{waiter_name}`'):
+            try:
+                self._client_wrapper.waiter.wait(StackName=self._name)
+                return True
+            except botocore.exceptions.WaiterError:
+                self.logger.error(f'Operation failed after waiting!')
+                return False
+    
+    def delete(self):
+        try:
+            self._client.delete_stack(StackName=self._name)
+            self.logger.info(f'Stack deletion triggered. Calling `stack_delete_complete` waiter!')
+            self.wait(waiter_name='stack_delete_complete')
+        except botocore.exceptions.ClientError as exp:
+            raise StackDeletionFailed(f'Stack deletion failed with exception {exp}')
+        except Exception as exp:
+            raise StackDeletionFailed(f'Stack deletion failed with exception {exp}')
+            
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        try:
+            self.logger.info(f'Exiting CFNStack context. Deleting the stack with name `{self._name}`, if it exists!')
+            if not self.status:
+                self.logger.info(f'Nothing to do, as stack doesn\'t exist.')
+                return
+            self.delete()
+        except Exception as exp:
+            self.logger.error('Please make sure stack gets deleted by checking the AWS console..')
+            raise StackDeletionFailed(f'Stack deletion failed with exception {exp}')
```

Comment:
I would add the comment: `Please make sure stack gets deleted by checking the AWS console` in the exception in case it is handled at some other level. I guess if we fail here the only way to work with it is to manually do some work